mypyc/genops.py

```diff
@@ -3388,7 +3398,8 @@ def add_bool_branch(self, value: Value, true: BasicBlock, false: BasicBlock) ->
             length = self.primitive_op(list_len_op, [value], value.line)
             zero = self.add(LoadInt(0))
             value = self.binary_op(length, zero, '!=', value.line)
-        elif isinstance(value.type, RInstance) and value.type.class_ir.has_method('__bool__'):
+        elif (isinstance(value.type, RInstance) and value.type.class_ir.is_ext_class
```

Comment:
Probably we'll stick with "yes" for now, but there is maybe a question about whether non-extension classes should be represented with RInstance at all. Maybe there should be different subclasses? Unclear.

Reply:
This would probably be helpful in the future with the immediate added benefit of not having to add this ext_class check alongside any RInstance checks, which seems easy to forget to do.